File: commons/test-utils/src/test/java/com/cedarsoft/test/utils/ParamTestTest.java
package com.cedarsoft.test.utils;

import static org.assertj.core.api.Assertions.*;

import javax.annotation.Nonnull;

import org.junit.jupiter.api.*;
import org.junit.jupiter.params.*;
import org.junit.jupiter.params.provider.*;

/**
 * @author Johannes Schneider (<a href="mailto:js@cedarsoft.com">js@cedarsoft.com</a>)
 */
@TestInstance(TestInstance.Lifecycle.PER_CLASS)
public class ParamTestTest {
  @ParameterizedTest
  @MethodSource("dataProvider")
  void daTest(@Nonnull String candidate, int val) {
    assertThat(Integer.parseInt(candidate)).isEqualTo(val);
  }

  @Nonnull
  public Arguments[] dataProvider() {
    return new Arguments[]{
      Arguments.of("1", 1),
      Arguments.of("2", 2),
      Arguments.of("3", 3),
      };
  }
}

File: commons/test-utils/src/test/kotlin/com/cedarsoft/test/utils/ParamTestTest.kt
package com.cedarsoft.test.utils

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.TestInstance
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource

/**
 */
@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class ParamTestTest {
  @ParameterizedTest
  @MethodSource("dataProvider")
  fun daTest(candidate: String, value: Int) {
    Assertions.assertThat(candidate.toInt()).isEqualTo(value)
  }

  fun dataProvider(): Array<Arguments> {
    return arrayOf(
      Arguments.of("1", 1),
      Arguments.of("2", 2),
      Arguments.of("3", 3)
    )
  }
}
